examples_tests/43.AllocatorTest/main.cpp

```diff
@@ -1,53 +1,219 @@
+constexpr size_t maxByteSizeForReservingMemory = 2147483648;	// 2GB
+constexpr size_t maxOffset = 512;
+constexpr size_t maxAlignOffset = 64;
+constexpr size_t maxBlockSize = 600;
+
+//TODO: full static
+class RandomNumberGenerator
+{
+public:
+	inline uint32_t getRndAllocCnt()    { return  allocsPerFrameRange(mt);  }
+	inline uint32_t getRndSize()        { return  sizePerFrameRange(mt); }
+	inline uint32_t getRndMaxAlign()    { return  (1u << maxAlignmentExpPerFrameRange(mt)); } //128 is max
+	inline uint32_t getRndBuffSize()    { return  buffSzRange(mt); }
+	inline uint32_t getRndOffset()      { return  offsetPerFrameRange(mt); }
+	inline uint32_t getRndAlignOffset() { return  alignOffsetPerFrameRange(mt); }
+	inline uint32_t getRndBlockSize()   { return  allocsPerFrameRange(mt); }
+
+	inline uint32_t getRandomNumber(uint32_t rangeBegin, uint32_t rangeEnd)   
+	{
+		std::uniform_int_distribution<uint32_t> dist(rangeBegin, rangeEnd);
+		return dist(mt);
+	}
+
+private:
+	std::mt19937 mt;
+	const std::uniform_int_distribution<uint32_t> allocsPerFrameRange = std::uniform_int_distribution<uint32_t>(minAllocs, maxAllocs);
+	const std::uniform_int_distribution<uint32_t> sizePerFrameRange = std::uniform_int_distribution<uint32_t>(1u, maxSizeForSquaring);
+	const std::uniform_int_distribution<uint32_t> maxAlignmentExpPerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxAlignmentExp);
+
+	const std::uniform_int_distribution<uint32_t> buffSzRange = std::uniform_int_distribution<uint32_t>(minByteSizeForReservingMemory, maxByteSizeForReservingMemory);
+	const std::uniform_int_distribution<uint32_t> offsetPerFrameRange = std::uniform_int_distribution<uint32_t>(0u, maxOffset);
+	const std::uniform_int_distribution<uint32_t> alignOffsetPerFrameRange = std::uniform_int_distribution<uint32_t>(0u, maxAlignOffset);
+	const std::uniform_int_distribution<uint32_t> blockSizePerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxBlockSize);
+};
+
+class AllocatorHandler
 {
-	//1
+public:
+	using PoolAddressAllocator = core::PoolAddressAllocatorST<uint32_t>;
+	using Traits = core::address_allocator_traits<PoolAddressAllocator>;
+	
+	AllocatorHandler(AllocationCreationParameters& allocationCreationParameters)
+		: creationParameters(allocationCreationParameters) 
 	{
-		void* resSpc = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(uint32_t), 16u, 1u), _IRR_SIMD_ALIGNMENT);
-		assert(resSpc != nullptr);
-		auto alctr = core::GeneralpurposeAddressAllocator<uint32_t>(resSpc, 0u, 0u, alignof(uint32_t), 16u, 1u);
-
-		for (int i = 0; i < 16u; i++)
-		{
-			auto addr = alctr.alloc_addr(1u, 1u);
-			if (addr == alctr.invalid_address)
-				os::Printer::print("invalid adress");
-			else
-				os::Printer::print(std::to_string(addr));
+	}
+	
+	//struct for holding temporary data (inputs of `multi_alloc_addr` i `multi_free_addr`)
+	struct EntryForFrameData
+	{
+		uint32_t outAddr[maxAllocs] = { ~0u };
```

Comment:
initialize all with `invalid_address`, the value ~0u is not that, `multi_alloc` wont work is the output array of addresses is not valid.